mitodl/mit-open
# Fix storybook github pages publishing

### What are the relevant tickets?
- closes https://github.com/mitodl/hq/issues/4760

### Description (What does it do?)
This PR:

1. fixes publishing storybook on github pages. It wasn't working before
    - it will only publish when manually run or when there are pushes to main.
2. Removes our `github-pages` react app.
    - At one point in the past, we had a react app + storybook deploying to github pages. The react app didn't have much besides a link to storybook, so we've just removed it.
3. Renames a few storybook stories.
    - Previously we had two "Categories": `ol-components/storyname` and `smoot-design/storyname`. I've moved most things under `ol-components/` to `smoot-design/` (this only affects the storybook, not the workspace name) and renamed `ol-components` category to "`old`"

### How can this be tested?
1. Run `yarn storybook` locally and compare with https://mitodl.github.io/mit-open/. They should be the same.
